scripts/core/exp-bindless-images.yml

```diff
@@ -0,0 +1,473 @@
+returns:
+    - $X_RESULT_ERROR_INVALID_CONTEXT
+    - $X_RESULT_ERROR_INVALID_VALUE
+    - $X_RESULT_ERROR_INVALID_MEM_OBJECT
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Allocate memory for bindless images"
+class: $xBindlessImages
+name: ImageAllocateExp
+ordinal: "0"
+analogue:
+    - "**cuArray3DCreate**"
+    - "**cuMipmappedArrayCreate**"
+params:
+    - type: $x_context_handle_t
+      name: hContext
+      desc: "[in] handle of the context object"
+    - type: "const $x_image_format_t*"
+      name: pImageFormat
+      desc: "[in] pointer to image format specification"
+    - type: "const $x_image_desc_t*"
+      name: pImageDesc
+      desc: "[in] pointer to image description"
+    - type: $x_exp_image_mem_handle_t*
+      name: phImageMem
+      desc: "[out] pointer to handle of image memory allocated"
+returns:
+    - $X_RESULT_ERROR_INVALID_CONTEXT
+    - $X_RESULT_ERROR_INVALID_VALUE
+    - $X_RESULT_ERROR_INVALID_IMAGE_FORMAT_DESCRIPTOR:
+      - "`pImageDesc && UR_MEM_TYPE_IMAGE1D_BUFFER < pImageDesc->type`"
+    - $X_RESULT_ERROR_INVALID_IMAGE_SIZE
+    - $X_RESULT_ERROR_INVALID_OPERATION
+    - $X_RESULT_ERROR_OUT_OF_RESOURCES
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Free memory for bindless images"
+class: $xBindlessImages
+name: ImageFreeExp
+ordinal: "0"
+analogue:
+    - "**cuArrayDestroy**"
+params:
+    - type: $x_context_handle_t
+      name: hContext
+      desc: "[in] handle of the context object"
+    - type: $x_exp_image_mem_handle_t
+      name: hImageMem
+      desc: "[in] handle of image memory to be freed"
+returns:
+    - $X_RESULT_ERROR_INVALID_CONTEXT
+    - $X_RESULT_ERROR_INVALID_VALUE
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Create a bindless unsampled image handle"
+class: $xBindlessImages
+name: UnsampledImageCreateExp
+ordinal: "0"
+analogue:
+    - "**cuSurfObjectCreate**"
+params:
+    - type: $x_context_handle_t
+      name: hContext
+      desc: "[in] handle of the context object"
+    - type: $x_exp_image_mem_handle_t
+      name: hImageMem
+      desc: "[in] handle to memory from which to create the image"
+    - type: "const $x_image_format_t*"
+      name: pImageFormat
+      desc: "[in] pointer to image format specification"
+    - type: "const $x_image_desc_t*"
+      name: pImageDesc
+      desc: "[in] pointer to image description"
+    - type: $x_mem_handle_t*
+      name: phMem
+      desc: "[out] pointer to handle of image object created"
+    - type: $x_exp_image_handle_t*
+      name: phImage
+      desc: "[out] pointer to handle of image object created"
+returns:
+    - $X_RESULT_ERROR_INVALID_CONTEXT
+    - $X_RESULT_ERROR_INVALID_VALUE
+    - $X_RESULT_ERROR_INVALID_IMAGE_FORMAT_DESCRIPTOR:
+      - "`pImageDesc && UR_MEM_TYPE_IMAGE1D_BUFFER < pImageDesc->type`"
```

Comment:
Maybe can be removed.

Reply:
Will treat all of these comments as the one above. Leaving for now, will remove if it is unnecessary.